### PR TITLE
feat: add e2e tests

### PR DESCRIPTION
**Pull request by wama-tw**

### CODE REVIEW

File: backend/src/app.module.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ import { ReportModule } from './report/report.module';
     CollegeModule,
     DepartmentModule,
     CourseModule,
-    AuthModule,
+    // AuthModule,
```
Comment (ppodds) [inline backend/src/app.module.ts:34]: Don't do this.

File: backend/src/seeder/course-seeder/course-seeder.service.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    course.teachers = '["梁德容", "王尉任", "鄭永斌"]';
+    course.classTimes = '["1-2", "1-3", "1-4"]';
+    course.limitCnt = 456;
+    course.admitCnt = 789;
+    course.waitCnt = 1011;
+    course.collegeId = 'collegeId';
+    course.departmentId = 'csie';
+    await this.courseRepository.save(course);
+
+    for (let i = 0; i < 200; i++) {
```
Comment (ppodds) [inline backend/src/seeder/course-seeder/course-seeder.service.ts:108]: Do you need so much fake data?

File: cypress.config.ts [hunk 3]
```diff
@@ -6,5 +6,6 @@ export default defineConfig({
     setupNodeEvents(on, config) {
       // implement node event listeners here
     },
+    baseUrl: "http://localhost:3001",
```
Comment (ppodds) [inline cypress.config.ts:9]:
```suggestion
    baseUrl: "http://localhost:4000",
```

File: backend/src/seeder/course-seeder/course-seeder.service.ts [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+@Injectable()
+export class CourseSeederService {
+  constructor(
+    @InjectRepository(Course)
+    private readonly courseRepository: Repository<Course>,
+  ) {}
+
+  async seedCourse() {
+    await this.courseRepository.clear();
```
Comment (ppodds) [inline backend/src/seeder/course-seeder/course-seeder.service.ts:14]: Could you extract it into the npm script?

**Review by ppodds — CHANGES_REQUESTED** [review]

File: backend/src/seeder/course-seeder/course-seeder.service.ts (hunk 2 above)
Comment (juliaouo) [reply backend/src/seeder/course-seeder/course-seeder.service.ts:108]: Because I need to test change the courses pages.

Comment (ppodds) [reply backend/src/seeder/course-seeder/course-seeder.service.ts:108]: Would the test pass if we only have 100 courses? You should consider the minimal test data.

Comment (wama-tw) [reply backend/src/seeder/course-seeder/course-seeder.service.ts:108]: I can help that

**Pull request merged**